pkg/sloop/ingress/kubewatcher.go

```diff
@@ -306,14 +312,32 @@ func (i *kubeWatcherImpl) processUpdate(kind string, obj interface{}, watchResul
 	glog.V(99).Infof("processUpdate: obj json: %v", resourceJson)
 
 	kubeMetadata, err := kubeextractor.ExtractMetadata(resourceJson)
+	EventInfo, err1 := kubeextractor.ExtractEventInfo(resourceJson)
+	InvolvedObject, err2 := kubeextractor.ExtractInvolvedObject(resourceJson)
 	if err != nil || kubeMetadata.Namespace == "" {
 		// We are only grabbing namespace here for a prometheus metric, so if metadata extract fails we just log and continue
 		glog.V(2).Infof("No namespace for resource: %v", err)
 	}
-	metricIngressKubewatchcount.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace).Inc()
-	metricIngressKubewatchbytes.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace).Add(float64(len(resourceJson)))
+	if err1 != nil {
+		glog.V(2).Infof("Extract event info: %v", err1)
+	}
+	if err2 != nil {
+		glog.V(2).Infof("Error occured while extracting Involved Object Info: %v", err2)
+	}
+	if enableGranularmetrics {
+
+		if kind == "Event" {
```

Comment:
[Minor] The check can be put together as 
if enableGranularmetrics & kind == "Event"
    then -- 
else 
     --